api/src/main/java/marquez/graphql/GraphqlDaos.java

```diff
@@ -163,87 +178,90 @@
+                   from jobs j_of_in
+                   left outer join job_versions_io_mapping io_of_in on io_of_in.job_version_uuid = j_of_in.current_version_uuid
+                    and io_of_in.io_type = 'INPUT'
+                   group by io_of_in.dataset_uuid
+              ) i on i.dataset_uuid = io_out.dataset_uuid
+          -- input jobs for each input dataset
+              left outer join (
+                  select io_of_out.dataset_uuid, jsonb_agg((select x from (select j_of_out.name, j_of_out.namespace_name as namespace) as x)) as out_agg
+                  from jobs_view j_of_out
+                  left outer join job_versions_io_mapping io_of_out
+                    on io_of_out.job_version_uuid = j_of_out.current_version_uuid
+                    and io_of_out.io_type = 'OUTPUT'
+                  group by io_of_out.dataset_uuid
+              ) o on o.dataset_uuid = io_out.dataset_uuid
+              WHERE io_out.io_type = 'OUTPUT'
+              group by io_out.job_version_uuid
+
+          ) d_in on d_in.job_version_uuid = j.current_version_uuid
+          --output datasets
+          left outer join(
+              select io_out.job_version_uuid, jsonb_agg((SELECT x FROM (SELECT ds_in.name, ds_in.namespace_name as namespace, o.out_agg as "inEdges", i.in_agg as "outEdges") AS x)) as agg
+              from job_versions_io_mapping io_out
+              inner join datasets ds_in on ds_in.uuid = io_out.dataset_uuid
+              -- output jobs for each output dataset
+              left outer join (
+                   select io_of_in.dataset_uuid, jsonb_agg((select x from (select j_of_in.name, j_of_in.namespace_name as namespace) as x)) as in_agg
+                   from jobs_view j_of_in
+                   left outer join job_versions_io_mapping io_of_in on io_of_in.job_version_uuid = j_of_in.current_version_uuid
+                    and io_of_in.io_type = 'INPUT'
+                   group by io_of_in.dataset_uuid
+              ) i on i.dataset_uuid = io_out.dataset_uuid
+              -- input jobs for each output dataset
+              left outer join (
+                  select io_of_out.dataset_uuid, jsonb_agg((select x from (select j_of_out.name, j_of_out.namespace_name as namespace) as x)) as out_agg
+                  from jobs_view j_of_out
+                  left outer join job_versions_io_mapping io_of_out
+                    on io_of_out.job_version_uuid = j_of_out.current_version_uuid
+                    and io_of_out.io_type = 'OUTPUT'
+                  group by io_of_out.dataset_uuid
+              ) o on o.dataset_uuid = io_out.dataset_uuid
+              WHERE io_out.io_type = 'INPUT'
+              group by io_out.job_version_uuid
+          ) d_out on d_out.job_version_uuid = j.current_version_uuid
+  """)
   List<JobResult> getLineage(String jobName, String namespaceName, Integer depth);
```

Comment:
Thanks for also updating the graphql SQL!

Reply:
I know, was visually looking to see if you missed anything 👀

Reply:
(GraphQL is also in beta, but to move it out of beta we'd need a proper test suite).